Mechanisms_evolution/Mechanisms_Evolution.Rmd

```diff
@@ -9,252 +9,222 @@ output:
+**These are column names. You will use them to analyze and plot the data.** 
 
-```{r dp, warning=FALSE, message=FALSE}
-# Load the dplyr library to provide a method of filtering the data
-library(dplyr)
-```
+* `exp` is the experiment number
+* `group` is your group number
+* `generation` is the number of generations from the beginning (0,1,2,3)
+* `snailcolor` is camouflaged or obvious (white or red)
+* `snails` is the number of snails
 
-Now filter the data.
 
-```{r}
-# Filter data: experiment is equal to 1
-exp1 <- filter(snail_data, exp==1)
-```
-
-This code filters for the snails in experiment 1.
-The `==` checks for equality (a `=` is the same as `<-` and assigns a value to a variable - that is not what you want here).
-
-We actually want to take the average of the class data for each experiment / generation / color combination. 
-
-* Get the rows of data for white snails in experiment 1 in generation 0.
-
-```{r}
-# Filter data: experiment 1, gen 0, col white
-exp1gen0colW <- filter(snail_data, exp==1 & generation==0 & snailcolor=="white")
-```
-
-* When you use words in R (e.g. "white"), put the word in quotes. Words that are not in quotes are interpreted as variables. There is no 'white' variable.
-* Calculate the average (mean) number of snails for these rows
-
-```{r , eval=FALSE}
-# Calculate mean of the snails column of exp1gen0colW
-mean(exp1gen0colW$snails)
-```
+## 3. Find the average snail count in groups of data
 
-Here we asking R to give us the mean of the subset of snails we found earlier. The `$` in the code points R to the the column of our data that we are interested in. 
+We are interested in differences in population size between red and white snails over time (`snailcolor`), between different generations (`generation`), and between experiments (`exp`). To graph the average snail population size over time, you first need to calculate the averages for each snail type at each generation time for each experiment. We want to take the average of the data for each experiment / generation / color combination. 
 
-We could keep doing this for each experiment, at each generation time, for each color of snail. Now I bet you're saying "Wait...there are 4 generation times, two colors, and two experiments, which would mean 16 lines of code!...this will take forever!", and you'd be correct! Let's let the computer do all this calculating for us.
-
-## 4. Automatically perform calculations on groups of data
-
-Rather than calculate the mean for each group manually we will use the `group_by` function from `dplyr` to split the data into groups. `group_by` takes the data, followed by each column you want to group by.
+We will use the `group_by` function from the `dplyr` package to create new variables with subsetted data. Tell `group_by` what variable you want to subset, followed by the columns you want to include.
```

Comment:
I have a feeling group_by will be complicated. I'm okay with not giving the example of individual subsets, but I would give a one variable group_by before a 3 variable. Also state an example of one group - eg exp1gen3colW so they know exactly what they are doing conceptually.

Reply:
never mind